binsync/ui/history_display/history_window.py

```diff
@@ -0,0 +1,178 @@
+            #return self.data_tooltips[row]
+            pass
+        return None
+
+    # No update_table because we don't care about the current state
+
+class HistoryTableView(BinsyncTableView):
+    HEADER = ['Addr', 'Function']
+
+    def __init__(self, controller: BSController, stretch_col=None,
+                 col_count=None, parent=None):
+        super().__init__(controller, None, 1, 2, parent)
+
+        self.model = HistoryTableModel(controller, self.HEADER, addr_col=0,
+                                        parent=parent)
+        self.proxymodel.setSourceModel(self.model)
+        self.setModel(self.proxymodel)
+
+        # always init settings *after* loading the model
+        self._init_settings()
+
+    def contextMenuEvent(self, event):
+        menu = QMenu(self)
+        menu.setObjectName("binsync_history_table_context_menu")
+
+        valid_row = True
+        selected_row = self.rowAt(event.pos().y())
+        idx = self.proxymodel.index(selected_row, 0)
+        idx = self.proxymodel.mapToSource(idx)
+        if event.pos().y() == -1 and event.pos().x() == -1:
+            idx = self.proxymodel.index(0, 0)
+            idx = self.proxymodel.mapToSource(idx)
+        elif not (0 <= selected_row < len(self.model.row_data)) or not idx.isValid():
+            valid_row = False
+
+        col_hide_menu = menu.addMenu("Show Columns")
+        handler = lambda ind: lambda: self._col_hide_handler(ind)
+        for i, c in enumerate(self.HEADER):
+            act = QAction(c, parent=menu)
+            act.setCheckable(True)
+            act.setChecked(self.column_visibility[i])
+            act.triggered.connect(handler(i))
+            col_hide_menu.addAction(act)
+
+        menu.popup(self.mapToGlobal(event.pos()))
+
+class HistoryDisplayWidget(QDialog):
+    def __init__(self,controller:BSController=None,parent=None):
+        super().__init__(parent)
+        self.controller = controller
+        self._init_widgets()
+        self._display_diff()
+        
+    def _init_widgets(self):
+        self.setWindowTitle("History")
+        
+        main_layout = QVBoxLayout()
+        top_layout = QHBoxLayout()
+        bottom_layout = QVBoxLayout()
+        
+        top_layout.addWidget(QLabel("Functions Changed in the Past Day"))
+        
+        self.table_view = HistoryTableView(self.controller)
+        bottom_layout.addWidget(self.table_view)
+        
+        
+        main_layout.addLayout(top_layout)
+        main_layout.addLayout(bottom_layout)
+        
+        self.setLayout(main_layout)
+        self.resize(1000, 800)
+        
+    def _display_diff(self):
+        changed_functions = []
+        client = self.controller.client
+        previous_time =  (datetime.now(timezone.utc)-timedelta(days=1)).timestamp()
+        old_commit = client.find_commit_before_ts(client.repo, previous_time,user_name=client.master_user)
+        old_state = client.parse_state_from_commit(client.repo,commit_hash=old_commit)
```

Comment:
Similar to the above scenario, I actually don't know how this ever worked... It's surprising because `parse_state_from_commit` actually does `repo.git.checkout(checkout_ref)`, but at the same time the Git-thread for updates is rapidly checkingout, so these should hit a race pretty fast.